app/helpers/application_helper/button/host_feature_button.rb

```diff
@@ -2,7 +2,19 @@ class ApplicationHelper::Button::HostFeatureButton < ApplicationHelper::Button::
   needs :@record
 
   def visible?
-    return false if @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
+    if @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
+      return true if ['start', 'stop'].include?(@feature.to_s)
+      return false
+    end
+
+    if @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager::Host)
+      case @feature.to_s
+        when 'start'
+          return @record.validate_start[:available]
```

Comment:
@romanblanco thanks for the detailed guidance, I really appreciate that :smile: 
But in this case its good to use the AvailabilityMixin and is_available methods, because we want to refactor one feature at a time and completely in on PR.

So I see two options: 
1. create a PR for the `start` feature and on for the `stop` feature and have them merged and then use the `supports_start?` methods accordingly in this PR.
2. use the `validate` methods here, get it merged and then do 1.

I would of course prefer 1. :smile_cat: but given time constraints, you probably want to do 2.